File: vca-backend-spring-boot-gp/src/main/java/com/vca/app/entity/Segment.java
package com.vca.app.entity;

import java.util.Date;

import jakarta.persistence.*;

@Entity
@Table(name = "segments")
public class Segment {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    @Column(nullable = false, unique = true)
    private String segName;

    // Define 'createdAt' and 'updatedAt' fields with 'TIMESTAMP' data type
    @Temporal(TemporalType.TIMESTAMP)
    @Column(nullable = false, updatable = false, columnDefinition = "TIMESTAMP DEFAULT CURRENT_TIMESTAMP")
    private Date createdAt;

    @Temporal(TemporalType.TIMESTAMP)
    @Column(nullable = false, columnDefinition = "TIMESTAMP DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP")
    private Date updatedAt;
    
    public Segment() {
    }
    
    public Segment(String segName) {
        this.segName = segName;
    }

    public long getId() {
        return id;
    }

    public String getName() {
        return segName;
    }

    @Override
    public String toString() {
        return "Segment [id=" + id + ", name=" + segName + "]";
    }
}

File: vca-backend-dot-net-gp-Final/Models/Segment.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace VCA.Models
{
    [Table("segments")]
    public class Segment
    {
        [Key]
        [Column("id")]
        public int Id { get; set; }

        [Required]
        [MaxLength(255)]
        [Column("seg_name")] // Use 'unique: true' for unique constraint
        public string? Name { get; set; }

        [DatabaseGenerated(DatabaseGeneratedOption.Computed)]
        public DateTime created_at { get; set; }

        [DatabaseGenerated(DatabaseGeneratedOption.Computed)]
        public DateTime updated_at { get; set; }

    }
}
